include/geos/io/GeoJSONReader.h

```diff
@@ -22,7 +22,12 @@
 #include <geos/geom/CoordinateSequence.h>
 #include <geos/geom/Geometry.h>
 #include <string>
-#include <json.hpp>
+
+#ifdef GEOS_COMPILATION
```

Comment:
hum, if this header is usable outside of GEOS internals, we can't use a different namespace between GEOS compilation and use of GEOS.
The only viable alternative is:
- don't use a vendorized nlohmann/json, but a system one
- or if using a vendorized nlohmann/json, modify the vendorized header to use a dedicated namespace. Using the #define trick would be dangerous for people using another nlohmann/json and GEOS headers at the same time

Reply:
Vendorize and namespace is the way. You can install into ``#include <geos/vend/json.hpp`` or some such. Many projects use nlohmann and are likely to clash if GEOS installed it in anywhere but it's own private spot and privately used it and did not expose it in its public headers.

Reply:
Really appreciate the feedback and discussion.  I am a little confused about what I should do.  So, I should move json.hpp to include/geos/vend/json.hpp and then create another file called include_nlohmann_json.hpp with something like:

```
#ifndef GEOS_VEND_JSON
#define GEOS_VEND_JSON
namespace geos {
    namespace vend {
        #include "json.hpp"
    }
}
#endif
```

And then use `#include "geos/vend/include_nlohmann_json.hpp"` in all other header files?

Reply:
Or is this enough?

```
#define nlohmann geos_nlohmann
#include "geos/vend/json.hpp"
```

Reply:
I don't think what you suggest above would work since json.hpp includes various headers, and that wouldn't be safe to do so in a namespace.

I think include_nlohmann_json.hpp should #undef nlohmann after #include "nlohmann/json.hpp"  , and then you you use explictly the geos_nlohmann namespace.
And for extra safety in case other folks before us would have played with #define nlohmann in a non-transient way:
```
#ifdef nlohmann
  #error "GEOS modifies the nlohmann define "
#endif
#define nlohmann geos_nlohmann
#include "nlohmann/json.hpp" 
#undef nlohmann
```

Reply:
I'm probably missing something obvious, but doesn't this work? (I didn't try it)

```
// normal includes
#include <string>

namespace geos {
#include "nlohmann/json.hpp"

namespace io {

class GeoJSONReader {
 // stuff
};

}
}
```

Reply:
Definitely does not work.

Reply:
Changes made.  Thanks again Even and Dan.  I learned something new today!